api/src/main/java/javax/enterprise/inject/spi/builder/CDIContainerBuilder.java

```diff
@@ -0,0 +1,369 @@
+     */
+    CDIContainerBuilder addAlternative(Class<?> alternativeClass);
+
+    /**
+     * Select alternative stereotypes for a synthetic bean archive.
+     *
+     * @param alternativeStereotypeClasses alternatives stereotypes to select
+     * @return self
+     */
+    CDIContainerBuilder alternativeStereotypes(Class<? extends Annotation>... alternativeStereotypeClasses);
+
+    /**
+     * Add an alternative stereotype class to the list of selected alternative stereotypes for a synthetic bean archive.
+     *
+     * @param alternativeStereotypeClass alternative stereotype to add
+     * @return self
+     */
+    CDIContainerBuilder addAlternativeStereotype(Class<? extends Annotation> alternativeStereotypeClass);
+
+    /**
+     * Set the configuration property for the container
+     *
+     * @param key property name
+     * @param value property value
+     * @return self
+     */
+    CDIContainerBuilder property(String key, Object value);
+
+    /**
+     * Set all the configuration properties.
+     *
+     * @param properties a map containing properties to add
+     * @return self
+     */
+    CDIContainerBuilder properties(Map<String, Object> properties);
+
+    /**
+     * Returns a {@link BeanConfigurator} intialized with the provided type
+     * to help you define an additional bean to add to the deployment.
+     *
+     * Bean will be created when initialize is called
+     *
+     * @param type type of the bean
+     * @return a BeanConfigurator ro define a custom bean
+     */
+    <T> BeanConfigurator<T> configureBean(Class<T> type);
+
+
+    /**
+     * Add the provided beans to set of discovered beans.
+     *
+     * @param beans beans to add
+     * @return self
+     */
+    CDIContainerBuilder AddBeans(Bean<?>... beans);
+
+    /**
+     *
+     * Add the provided bean to set of discovered beans.
+     *
+     * @param bean bean to add
+     * @return self
+     */
+    CDIContainerBuilder addBean(Bean<?> bean);
+
+
+    /**
+     * @return self
+     * @see #disableDiscovery()
+     */
+    CDIContainerBuilder enableDiscovery();
+
+    /**
+     * By default, the discovery is enabled. However, it's possible to disable the discovery completely so that only the "synthetic" bean archive is considered.
+     *
+     * @return self
+     */
+    CDIContainerBuilder disableDiscovery();
+
+    /**
+     * @return <code>true</code> if the discovery is enabled, <code>false</code> otherwise
+     * @see #disableDiscovery()
+     */
+    boolean isDiscoveryEnabled();
+
+    /**
+     * Set a {@link ClassLoader}. The given {@link ClassLoader} will be scanned automatically for bean archives if scanning is enabled.
+     *
+     * @param classLoader the class loader to use
+     * @return self
+     */
+    CDIContainerBuilder setClassLoader(ClassLoader classLoader);
+
+    /**
+     * <p>
+     * Initializes a CDI Container.
+     * </p>
+     * <p>
+     * Cannot be called within an application server.
```

Comment:
Not this method but the whole `CDIContainerBuilder` class should not be used in EE, right?